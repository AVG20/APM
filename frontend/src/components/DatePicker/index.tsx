import generatePicker from 'antd/es/date-picker/generatePicker';
import dayjsGenerateConfig from 'antd/node_modules/rc-picker/lib/generate/dayjs';

const DatePicker = generatePicker(dayjsGenerateConfig);

export default DatePicker;
