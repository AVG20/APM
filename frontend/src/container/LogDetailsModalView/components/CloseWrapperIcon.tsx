import { CloseOutlined, EditFilled } from '@ant-design/icons';
import { MouseEventHandler } from 'react';

import ICloseWrapperIcon from '../interfaces/ICloseWrapperIcon';

function CloseWrapperIcon({ toggleInput }: ICloseWrapperIcon): JSX.Element {
	const handleClick: MouseEventHandler<Element> = (event) => {
		event.preventDefault();
		event.stopPropagation();
		toggleInput();
	};

	return (
		<div style={{ display: 'flex', gap: 20, marginLeft: -33 }}>
			<span role="button" aria-hidden="true" onClick={handleClick}>
				<EditFilled />
			</span>
			<CloseOutlined />
		</div>
	);
}

export default CloseWrapperIcon;
