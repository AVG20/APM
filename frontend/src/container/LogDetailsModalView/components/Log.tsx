import { Tooltip, Typography } from 'antd';
import dayjs from 'dayjs';
import { useMemo } from 'react';

import ILog from '../interfaces/ILog';
import { LogContainer, LogDiv } from '../styles/Log';

const { Text } = Typography;

function Log({ log }: ILog): JSX.Element {
	const text = useMemo(
		() => `${dayjs(log.timestamp / 1e6).format()} | ${log.body}`,
		[log],
	);

	return (
		<LogContainer>
			<Tooltip title={text}>
				<LogDiv>
					<Text ellipsis>{text}</Text>
				</LogDiv>
			</Tooltip>
		</LogContainer>
	);
}

export default Log;
