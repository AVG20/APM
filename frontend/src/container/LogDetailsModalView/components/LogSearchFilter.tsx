import { Input, InputRef, Popover } from 'antd';
import SearchFields from 'container/LogsSearchFilter/SearchFields';
import { useCallback, useRef, useState } from 'react';

import { Container, DropDownContainer } from '../../LogsSearchFilter/styles';

interface ILogSearchFilter {
	query: string;
	setQuery: (value: string) => void;
}

function LogSearchFilter({ query, setQuery }: ILogSearchFilter): JSX.Element {
	// const [searchText, setSearchText] = useState('');
	const [showDropDown, setShowDropDown] = useState(false);
	const searchRef = useRef<InputRef>(null);

	const onDropDownToggleHandler = useCallback(
		(value: boolean) => (): void => {
			setShowDropDown(value);
		},
		[],
	);

	const onPopOverChange = useCallback(
		(isVisible: boolean) => {
			onDropDownToggleHandler(isVisible)();
		},
		[onDropDownToggleHandler],
	);

	return (
		<Container>
			<Popover
				placement="bottom"
				content={
					<DropDownContainer>
						<SearchFields
							updateQueryString={setQuery}
							onDropDownToggleHandler={onDropDownToggleHandler}
						/>
					</DropDownContainer>
				}
				trigger="click"
				overlayInnerStyle={{
					minWidth: 800,
					width: `${searchRef?.current?.input?.offsetWidth || 0}px`,
				}}
				open={showDropDown}
				destroyTooltipOnHide
				onOpenChange={onPopOverChange}
			>
				<Input.Search
					ref={searchRef}
					placeholder="Search Filter"
					value={query}
					onChange={(e): void => {
						const { value } = e.target;
						setQuery(value);
					}}
					onSearch={setQuery}
					allowClear
				/>
			</Popover>
		</Container>
	);
}

export default LogSearchFilter;
