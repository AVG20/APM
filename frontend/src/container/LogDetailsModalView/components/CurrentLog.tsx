import { Tooltip, Typography } from 'antd';
import dayjs from 'dayjs';
import { useMemo } from 'react';

import { ICurrentLog } from '../interfaces/ILog';
import { LogContainer, LogDiv } from '../styles/Log';

const { Text } = Typography;

function CurrentLog({ log }: ICurrentLog): JSX.Element {
	const text = useMemo(
		() => `${dayjs(log.timestamp / 1e6).format()} | ${log.body}`,
		[log],
	);

	return (
		<LogContainer>
			<LogDiv>
				<Tooltip title={text}>
					<Text ellipsis>{text}</Text>
				</Tooltip>
			</LogDiv>
		</LogContainer>
	);
}

export default CurrentLog;
