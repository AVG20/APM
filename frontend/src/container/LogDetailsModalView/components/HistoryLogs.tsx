import { Button, Spin, Typography } from 'antd';

import IHistoryLogs, { HistoryPosition } from '../interfaces/IHistoryLogs';
import {
	HisoryLogContainer,
	HistoryLogHeader,
	HistoryLogsScrolled,
	LogContainer,
	LogDiv,
	SpinnerContainer,
} from '../styles/Log';
import Log from './Log';

const { Text } = Typography;

function HistoryLogs({
	logs,
	position,
	addMoreLogs,
	isLoad,
	isError,
}: IHistoryLogs): JSX.Element {
	if (isError) {
		return <div>Something went wrong</div>;
	}

	return (
		<>
			{position === HistoryPosition.prev && (
				<HistoryLogHeader>
					Showing 10 lines after match
					<Button size="small" onClick={addMoreLogs}>
						Show 10 more lines
					</Button>
				</HistoryLogHeader>
			)}
			<HisoryLogContainer>
				{isLoad ? (
					<SpinnerContainer>
						<Spin />
					</SpinnerContainer>
				) : (
					<HistoryLogsScrolled>
						{logs?.length === 0 ? (
							<LogContainer>
								<LogDiv>
									<Text ellipsis>There are no logs</Text>
								</LogDiv>
							</LogContainer>
						) : (
							logs?.map((log) => (
								<Log key={log.data.id} timestamp={log.timestamp} log={log} />
							))
						)}
					</HistoryLogsScrolled>
				)}
			</HisoryLogContainer>
			{position === HistoryPosition.next && (
				<HistoryLogHeader>
					Showing 10 lines before match
					<Button size="small" onClick={addMoreLogs}>
						Show 10 more lines
					</Button>
				</HistoryLogHeader>
			)}
		</>
	);
}

export default HistoryLogs;
