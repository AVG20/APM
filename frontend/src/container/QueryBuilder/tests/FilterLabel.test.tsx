import { render } from '@testing-library/react';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import i18n from 'ReactI18';
import store from 'store';

import { FilterLabel } from '../components';

describe('QueryBuilder', () => {
	const labelText = 'Filter label text';
	it('should render the label text', () => {
		const { getByText } = render(<FilterLabel label={labelText} />);
		const labelElement = getByText(labelText);
		expect(labelElement).toBeInTheDocument();
	});

	it('should render FilterLabel', () => {
		const { asFragment } = render(
			<MemoryRouter>
				<Provider store={store}>
					<I18nextProvider i18n={i18n}>
						<FilterLabel label={labelText} />
					</I18nextProvider>
				</Provider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
