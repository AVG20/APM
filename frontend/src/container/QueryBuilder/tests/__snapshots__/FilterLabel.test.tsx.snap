// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryBuilder: Filter Labels it should render FilterLabel 1`] = `
<DocumentFragment>
  .c0 {
  padding: 0 0.6875rem;
  min-height: 2rem;
  width: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.125rem;
  border: 0.0625rem solid #434343;
  background-color: #141414;
}

<div
    class="c0"
  >
    Filter label text
  </div>
</DocumentFragment>
`;
